diff --git a/dist/node/chunks/node.js b/dist/node/chunks/node.js
index 8d6dbaf1f21914bcf02576f8a5f13546a3dcdce8..49b30abd781d906c2cbc47e6c035f00f70b26554 100644
--- a/dist/node/chunks/node.js
+++ b/dist/node/chunks/node.js
@@ -35651,17 +35651,19 @@ const _require = createRequire(
 );
 async function loadConfigFromBundledFile(fileName, bundledCode, isESM) {
 	if (isESM) {
-		let nodeModulesDir = typeof process.versions.deno === "string" ? void 0 : findNearestNodeModules(path.dirname(fileName));
-		if (nodeModulesDir) try {
-			await fsp.mkdir(path.resolve(nodeModulesDir, ".vite-temp/"), { recursive: true });
+		const nodeModulesDir = typeof process.versions.deno === "string" ? void 0 : findNearestNodeModules(path.dirname(fileName));
+		let viteTempDir = nodeModulesDir ? path.resolve(nodeModulesDir, ".vite-temp") : void 0;
+		if (viteTempDir) try {
+			await fsp.mkdir(viteTempDir, { recursive: true });
 		} catch (e) {
-			if (e.code === "EACCES") nodeModulesDir = void 0;
+			if (e.code === "EACCES") viteTempDir = void 0;
 			else throw e;
 		}
 		const hash = `timestamp-${Date.now()}-${Math.random().toString(16).slice(2)}`;
-		const tempFileName = nodeModulesDir ? path.resolve(nodeModulesDir, `.vite-temp/${path.basename(fileName)}.${hash}.mjs`) : `${fileName}.${hash}.mjs`;
-		ignoreInput(tempFileName);
-		ignoreOutput(tempFileName);
+		const tempFileName = viteTempDir ? path.resolve(viteTempDir, `${path.basename(fileName)}.${hash}.mjs`) : `${fileName}.${hash}.mjs`;
+		const pathToIgnore = viteTempDir ?? tempFileName;
+		ignoreInput(pathToIgnore);
+		ignoreOutput(pathToIgnore);
 		await fsp.writeFile(tempFileName, bundledCode);
 		try {
 			return (await import(pathToFileURL(tempFileName).href)).default;
